src/content-scripts/components/RepoVoting/RepoUnvoteButton.ts

```diff
@@ -0,0 +1,42 @@
+import "../../content-scripts.css";
+import { createHtmlElement } from "../../../utils/createHtmlElement";
+import { getAuthToken } from "../../../utils/checkAuthentication";
+import { OPEN_SAUCED_REPOS_ENDPOINT } from "../../../constants";
+import { VoteRepoButton } from "./RepoVoteButton";
+
+export const RepoUnvoteButton = (ownerName: string, repoName: string) => {
+  const repoUnvoteButton = createHtmlElement("a", {
+    className:
+      "inline-block text-white rounded-md  text-sm  text-center select-none border hover:shadow-button hover:no-underline no-underline border-none cursor-pointer bg-gradient-to-r from-[#e67e22] to-[#d35400] py-1 px-2 mx-2",
+    innerHTML: `
+            <span>Unvote</span>
+            <svg stroke="currentColor" fill="currentColor" stroke-width="0" viewBox="0 0 24 24" height="1em" width="1em" xmlns="http://www.w3.org/2000/svg" class="align-middle"><g><path fill="none" d="M0 0h24v24H0z"></path><path d="M12 22C6.477 22 2 17.523 2 12S6.477 2 12 2s10 4.477 10 10-4.477 10-10 10zm-.997-6l7.07-7.071-1.414-1.414-5.656 5.657-2.829-2.829-1.414 1.414L11.003 16z"></path></g></svg>
+        `,
+  });
+
+  repoUnvoteButton.addEventListener("click", async () => {
+    repoUnvoteButton.innerHTML = `
+      <span>Loading...</span>
+    `;
+    const userToken = await getAuthToken();
+
+    const response = await fetch(
+      `${OPEN_SAUCED_REPOS_ENDPOINT}/${ownerName}/${repoName}/vote`,
+      {
+        method: "DELETE",
+        headers: { Authorization: `Bearer ${userToken}` },
+      },
+    );
```

Comment:
Can we have the vote API request as a utility function with the methods "PUT", "DELETE" parameterized? This would allow us to call this function from both the vote button and unvote button without duplicating code.
https://github.com/open-sauced/browser-extensions/blob/0589d54a4e3f4a8e39f6a5116c900932db89cc8d/src/content-scripts/components/RepoVoting/RepoVoteButton.ts#L21-L29

Reply:
Yup, makes sense. This should go in the api interaction file.